File: C/aoc_2021/day1/day1.c
/**
 * Author: Ethan Riegsecker
 *
 * Date: 1/5/2022
 *
 */

#include <stdlib.h>
#include <stdio.h>
#include <string.h>

#include "../../utils/file_utils/file_utils.h"

int main(int argc, char **argv) {
	char *filepath;
	if(argc == 2) {
		filepath = argv[1];
	} else {
		filepath = "in.txt";
	}

	int numLines = getNumLines(filepath);
	char **lines = getFileLines(filepath, numLines);
	int *num = (int *) malloc(sizeof(int) * numLines);
	for(int i=0; i<numLines; i++) {
		num[i] = atoi(lines[i]);
		free(lines[i]);
	}
	free(lines);

	int ans1 = 0;
	int ans2 = 0;
	for(int i=0; i<numLines; i++) {
		if(i >= 1 && num[i] > num[i-1]) {
			ans1++;
		}
	       	if(i >= 3 && (num[i] + num[i-1] + num[i-2]) > 
				(num[i-1] + num[i-2] + num[i-3])) {
			ans2++;
		}
	}

	printf("answer #1 = %d\nanswer #2 = %d\n",ans1 , ans2);
	return 0;

}

File: Python/aoc_2021/day1/day1.py
input = "in.txt"

ans1 = 0
ans2 = 0
XS = [int(x) for x in open(input)]
for i in range(len(XS)):
    if i >= 1 and XS[i] > XS[i-1]:
        ans1 += 1
    if i >= 3 and XS[i] > XS[i-3]:
        ans2 += 1

print("part 1 answer:", ans1)
print("part 2 answer:", ans2)
